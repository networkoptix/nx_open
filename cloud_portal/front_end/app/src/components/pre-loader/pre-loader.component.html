<div class="placeholder" [ngClass]="type">
    <div class="placeholder-content">
        <div class="placeholder-preloader">
            <div class="circleG circleG_1"></div>
            <div class="circleG circleG_2"></div>
            <div class="circleG circleG_3"></div>
        </div>
    </div>
</div>
