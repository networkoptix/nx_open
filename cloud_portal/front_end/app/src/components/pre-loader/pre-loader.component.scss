.placeholder-content {
    height    : 30px;
    display   : flex;
    transform : translateY(-50%);
}
