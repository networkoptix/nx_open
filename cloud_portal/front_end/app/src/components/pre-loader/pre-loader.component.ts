import { Component, Input, OnInit } from '@angular/core';

/* Usage
<nx-pre-loader
    type?="page">
</nx-pre-loader>
*/

@Component({
    selector: 'nx-pre-loader',
    templateUrl: 'pre-loader.component.html',
    styleUrls: []
})
export class NxPreLoaderComponent implements OnInit {
    @Input() type: string;

    constructor() {
    }

    ngOnInit() {
        this.type = this.type || '';
    }
}
