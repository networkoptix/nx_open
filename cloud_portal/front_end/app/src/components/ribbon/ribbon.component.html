<div class="clearfix ribbon" *ngIf="showRibbon">
    <div class="message">{{message}}</div>
    <div class="action"><a [href]="actionUrl">{{action}}</a></div>
</div>
