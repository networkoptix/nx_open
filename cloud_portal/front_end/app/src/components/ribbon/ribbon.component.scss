@import '../../../styles/custom/custom_palette';
@import '../../../styles/theme_palette';
@import '../../../styles/theme_variables';

div {
    background-color : $brand_d1;
    color            : $light1;
    line-height      : 1rem;
    font-family      : $font-family-medium;
    font-size        : $font-size-base;
    
    &.ribbon {
        border-top    : 1px solid $brand_d2;
        border-bottom : 1px solid $brand_d2;
        padding       : .5rem 1rem;
        display       : flex;
        align-items   : center;
        
        .message {
            flex : 70%;
        }
        
        .action {
            flex       : 30%;
            text-align : right;
        }
    }
    
    & > a {
        color           : $light1;
        font-family     : $font-family-regular;
        font-size       : $font-size-base;
        text-decoration : underline;
        
        &:hover {
            opacity : .8;
        }
    }
}
