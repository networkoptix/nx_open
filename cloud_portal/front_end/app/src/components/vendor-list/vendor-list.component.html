<nx-pre-loader *ngIf="vendors == undefined"></nx-pre-loader>

<div class="row pl-3 pr-3 pb-2 m-0 campage-info" [hidden]="filter.query" *ngIf="vendors && vendors.length">
    <div class="col-12 col-sm-6 row">
        <div class="col-12 col-lg-6 pl-0">
            <span>{{ vendors.length }} manufacturers</span>
        </div>
        <div class="col-12 col-lg-6 pl-0 pl-lg-2">
            <span>{{ cameras }} devices</span>
        </div>
    </div>

    <hr class="mb-0 mt-3 w-100">
</div>

<div class="row p-2 pb-5" *ngIf="vendors && !filter.query">
    <div class="col-12 col-sm-6 row">
        <div class="col-12 col-lg-6">
            <ul class="pl-2 m-0 list-unstyled">
                <li class="no-left-padding"
                    *ngFor="let vendor of vendors.slice(vendorGroup(0).start, vendorGroup(0).end)">
                    <a href (click)="setVendor(vendor); false">
                    <span>
                        {{ vendor }}
                    </span>
                    </a>
                </li>
            </ul>
        </div>
        <div class="col-12 col-lg-6">
            <ul class="pl-2 m-0 list-unstyled">
                <li class="no-left-padding"
                    *ngFor="let vendor of vendors.slice(vendorGroup(1).start, vendorGroup(1).end)">
                    <a href (click)="setVendor(vendor); false">
                    <span>
                        {{ vendor }}
                    </span>
                    </a>
                </li>
            </ul>
        </div>
    </div>
    <div class="col-12 col-sm-6 row">
        <div class="col-12 col-lg-6">
            <ul class="pl-2 m-0 list-unstyled">
                <li class="no-left-padding"
                    *ngFor="let vendor of vendors.slice(vendorGroup(2).start, vendorGroup(2).end)">
                    <a href (click)="setVendor(vendor); false">
                    <span>
                        {{ vendor }}
                    </span>
                    </a>
                </li>
            </ul>
        </div>
        <div class="col-12 col-lg-6">
            <ul class="pl-2 m-0 list-unstyled">
                <li class="no-left-padding"
                    *ngFor="let vendor of vendors.slice(vendorGroup(3).start, vendorGroup(3).end)">
                    <a href (click)="setVendor(vendor); false">
                    <span>
                        {{ vendor }}
                    </span>
                    </a>
                </li>
            </ul>
        </div>
    </div>
</div>
