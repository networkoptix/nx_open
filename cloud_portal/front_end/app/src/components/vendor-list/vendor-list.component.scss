@import '../../../styles/custom/custom_palette';
@import '../../../styles/theme_palette';
@import '../../../styles/theme_variables';

