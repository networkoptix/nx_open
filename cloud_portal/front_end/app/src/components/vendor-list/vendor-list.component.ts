import {
    Component, EventEmitter, forwardRef, Input, OnInit, Output
} from '@angular/core';
import { NxConfigService }   from '../../services/nx-config';
import { NG_VALUE_ACCESSOR } from '@angular/forms';

/* USAGE
 <nx-vendor-list
         vendors=[]
         cameras=[]
        [(ngModel)]="filterModel">
 </nx-vendor-list>
 */

@Component({
    selector   : 'nx-vendor-list',
    templateUrl: 'vendor-list.component.html',
    styleUrls  : ['vendor-list.component.scss'],
    providers: [{
        provide    : NG_VALUE_ACCESSOR,
        useExisting: forwardRef(() => NxVendorListComponent),
        multi      : true
    }]
})
export class NxVendorListComponent implements OnInit {
    @Input() vendors: any;
    @Input() cameras: any;

    CONFIG: any;
    arrGridColumns: number[];

    private filter: any = {};

    constructor(CONFIG: NxConfigService) {
        this.CONFIG = CONFIG.getConfig();

        this.arrGridColumns = Array.apply(undefined, Array(this.CONFIG.layout.vendorGroups)).map((x, i) => i);
    }

    ngOnInit() {}

    vendorGroup(n) {
        function nstart(m) {
            if (m === 0) {
                return 0;
            }

            return nstart(m - 1) + ncol(m - 1);
        }

        function ncol(m) {
            return m < (cols - (M * cols - total)) ? M : (M - 1);
        }

        const total = this.vendors.length,
              cols  = this.CONFIG.layout.vendorGroups,
              M     = Math.ceil(total / cols),
              n1    = nstart(n),
              n2    = ncol(n);

        return { start: n1, end: n1 + n2 };
    }

    // Form control functions
    // The method set in registerOnChange to emit changes back to the form
    private propagateChange = (_: any) => {};

    writeValue(value: any) {
        if (value) {
            this.filter = value;
        }
    }

    /**
     * Set the function to be called
     * when the control receives a change event.
     */
    registerOnChange(fn) {
        this.propagateChange = fn;
    }

    /**
     * Set the function to be called
     * when the control receives a touch event.
     */
    registerOnTouched(fn: () => void): void {
    }

    setVendor(value) {
        this.filter.query = value;
        // Propagate component's value attribute (model)
        this.propagateChange(this.filter);
    }

}
