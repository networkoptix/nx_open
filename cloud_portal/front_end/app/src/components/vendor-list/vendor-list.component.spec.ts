import { async, ComponentFixture, inject, TestBed } from '@angular/core/testing';

import { NxVendorListComponent } from './vendor-list.component';
import { NxPreLoaderComponent }  from '../pre-loader/pre-loader.component';
import { NxRibbonService }       from '../ribbon/ribbon.service';

describe('NxVendorListComponent', () => {
    let component: NxVendorListComponent;
    let fixture: ComponentFixture<NxVendorListComponent>;

    beforeEach(async(() => {
        TestBed.configureTestingModule({
            declarations: [NxVendorListComponent, NxPreLoaderComponent]
        }).compileComponents();
    }));

    beforeEach(() => {
        fixture = TestBed.createComponent(NxVendorListComponent);
        component = fixture.componentInstance;
        fixture.detectChanges();
    });

    it('should create', () => {
        expect(component).toBeTruthy();
    });

    it('renders correctly', () => {
        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();
    });
});
