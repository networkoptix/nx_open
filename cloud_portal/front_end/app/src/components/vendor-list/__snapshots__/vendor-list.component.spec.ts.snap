// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NxVendorListComponent renders correctly 1`] = `
<nx-vendor-list
  CONFIG={[Function Object]}
  filter={[Function Object]}
  propagateChange={[Function Function]}
  vendorGroups={[Function Number]}
>
  <nx-pre-loader>
    <div
      class="placeholder-wrapper"
      ng-reflect-ng-class="[object Object]"
    >
      <div
        class="placeholder"
        ng-reflect-klass="placeholder"
        ng-reflect-ng-class=""
      >
        <div
          class="placeholder-content"
        >
          <div
            class="placeholder-preloader"
          >
            <div
              class="circleG circleG_1"
            />
            <div
              class="circleG circleG_2"
            />
            <div
              class="circleG circleG_3"
            />
          </div>
        </div>
      </div>
    </div>
  </nx-pre-loader><div
    class="row pl-3 pb-3 m-0 campage-info"
  >
    <h4>
      
    </h4>
  </div>
</nx-vendor-list>
`;
